extensions/amp-lightbox/1.0/lightbox.js

```diff
@@ -0,0 +1,150 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../src/preact';
+import {ContainWrapper} from '../../../src/preact/component';
+import {forwardRef} from '../../../src/preact/compat';
+import {setStyle} from '../../../src/style';
+import {
+  useEffect,
+  useImperativeHandle,
+  useRef,
+  useState,
+} from '../../../src/preact';
+
+const ANIMATION_PRESETS = {
+  'fade-in': [{opacity: 0}, {opacity: 1}],
+  'fly-in-top': [
+    {opacity: 0, transform: 'translate(0,-100%)'},
+    {opacity: 1, transform: 'translate: (0, 0)'},
+  ],
+  'fly-in-bottom': [
+    {opacity: 0, transform: 'translate(0, 100%)'},
+    {opacity: 1, transform: 'translate: (0, 0)'},
+  ],
+};
+
+/**
+ * @param {!LightboxDef.Props} props
+ * @param {{current: (!LightboxDef.LightboxApi|null)}} ref
+ * @return {PreactDef.Renderable}
+ */
+function LightboxWithRef(
+  {
+    id,
+    layout = 'nodisplay',
+    animateIn = 'fade-in',
+    closeButtonAriaLabel,
+    // eslint-disable-next-line no-unused-vars
+    scrollable, // (TODO: discussion)
+    children,
+    onOpen,
+    openOnLoad,
+    ...rest
+  },
+  ref
+) {
+  const lightboxRef = useRef(null);
+  const [show, setShow] = useState(openOnLoad);
+
+  useImperativeHandle(
+    ref,
+    () => ({
+      open: () => {
+        setShow(true);
+      },
+      close: () => {
+        setShow(false);
+      },
+    }),
+    [setShow]
+  );
+
+  useEffect(() => {
+    const element = lightboxRef.current;
+    if (show) {
+      setStyle(element, 'visibility', 'visible');
+      element.animate(ANIMATION_PRESETS[animateIn], {duration: 200});
+      // setStyle(element, 'display', 'flex');
+      setStyle(element, 'opacity', 1);
+      if (onOpen) {
+        onOpen();
+      }
+    } else {
+      setStyle(element, 'visibility', 'hidden');
+      element.animate(ANIMATION_PRESETS[animateIn].reverse(), {duration: 200});
+      // setStyle(element, 'display', 'none');
+      setStyle(element, 'opacity', 0);
+    }
+  }, [show, animateIn, onOpen]);
+
+  return (
+    <ContainWrapper
+      {...rest}
+      ref={lightboxRef}
+      id={id}
+      layout={layout}
+      size={true}
+      layout={true}
+      paint={true}
+      style={{
+        zIndex: 1000,
+        backgroundColor: 'rgba(0, 0, 0, 0.9)',
+        color: '#fff',
+        width: window.innerWidth,
+        height: window.innerHeight,
+        opacity: 0,
+        visibility: 'hidden',
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'right',
+      }}
+    >
+      <button
+        textContent={closeButtonAriaLabel}
+        tabIndex={-1}
```

Comment:
Why not?

Reply:
This is screen-reader only `close` button, so it's meant to be hidden and not shown on the screen. Only someone navigating using a screen reader should be able to access and interact with this button so that they can close out of the lightbox when they want (since it is assumed they cannot tab out, `esc`, etc.)

Reply:
Ok. Understood. And the screen reader ignores `tabIndex=-1`?

Reply:
Yes, it seems like it does. This is taken from the old code so I'm assuming it was implemented with this in mind. However, I think `tabindex=-1` here to make sure that the button cannot be reached in any other way other than screen readers.